### PR TITLE
[MRG] add a `mle kaggle` cli, json mode and kaggle requirement summarize agent

### LINKED ISSUE #95: make kaggle a seperate cli and improve it to a chat assistant
#### Software execution information

MLE-agent version:
System OS version:

#### Other information

Things you tried, stack traces, related issues, suggestions on how to fix it...

### PR DESCRIPTION
**Pull request by HuaizhengZhang**
Closes #95

- [x] add `mle kaggle`
- [x] add kaggle requirement summarize agent
- [x] add `json mode` to the llm agent

next PR
- [ ] refactor the workflow as a chat assistant
- [ ] improve the search logic
- [ ] add more tool callings #2 

#### Before submitting this PR, please make sure you have:

- [x] confirmed all checks still pass OR confirm CI build passes.
- [x] verified that any code or assets from external sources are properly credited in comments and/or in
  the [credit file](https://github.com/MLSysOps/MLE-agent/blob/main/.github/OPEN_SOURCE_LICENSES.md).

### CODE REVIEW

File: agent/model/models.py [hunk 1]
```diff
@@ -77,15 +77,23 @@ def __init__(self, api_key, model, temperature):
...
+        json_mode
+
+        Returns
+        -------
+
         """
 
         completion = self.client.chat.completions.create(
             model=self.model,
+            response_format={"type": "json_object"} if json_mode else {"type": "text"},
```
Comment (huangyz0918) [inline agent/model/models.py:96]: this will destroy the Ollama support

File: agent/kaggle_cli.py [hunk 2]
```diff
@@ -0,0 +1,78 @@
+import os
```
Comment (huangyz0918) [inline agent/kaggle_cli.py:1]: no necessary to add a new cli enrty

File: agent/build_config.py [hunk 3]
```diff
@@ -0,0 +1,174 @@
+import sys
```
Comment (huangyz0918) [inline agent/build_config.py:1]: I don't understand 1. why build_config.py is a file 2.  why it is under the agent folder

File: agent/kaggle_cli.py (hunk 2 above)
Comment (HuaizhengZhang) [reply agent/kaggle_cli.py:1]: An experimental CLI to improve the kaggle workflow.

Don't want to influence the baseline mode and add too many if-else before finishing this workflow.

File: agent/build_config.py (hunk 3 above)
Comment (HuaizhengZhang) [reply agent/build_config.py:1]: 1. kaggle cli and baseline cli share this, so I move it to a single file for now.

2. Try to put it into utils/config, but meet the recurrent importing issues.

Conclusion 

After I rebuild the kaggle workflow,we need your help to improve the structure (maybe after your reproduce)

Comment (huangyz0918) [reply agent/build_config.py:1]: I will re-structure and refactor the whole project, no worries

File: agent/model/models.py (hunk 1 above)
Comment (HuaizhengZhang) [reply agent/model/models.py:96]: Let me fix this

**Pull request merged**